File: internal/model/generator.go
package model

type GeneratorFilter struct {
	WorkspaceID string
	PageSize    int
	PageNumber  int
	Query       string
}

type Generator struct {
	WorkspaceID string `json:"workspace_id"`
	ID          string `json:"id"`
	Name        string `json:"name"`
	Prompt      string `json:"prompt"`
	Provider    string `json:"provider"` // AI provider: openai, gemini, etc.
	Model       string `json:"model"`
	Modality    string `json:"modality"`
	ImageURLs   string `json:"image_urls"` // Comma-separated image URLs for template default images
	CreatedAt   string `json:"created_at"`
	CreatedBy   string `json:"created_by"`
	UpdatedAt   string `json:"updated_at"`
	UpdatedBy   string `json:"updated_by"`
}

// TableName specifies the table name for Generator
func (Generator) TableName() string {
	return "ai_generators"
}

File: web/src/types/generator.ts
export type Modality = 'text2text';

export interface Generator {
  id: string;
  workspace_id: string;
  name: string;
  prompt: string;
  provider: string; // AI provider: openai, gemini, etc.
  model: string;
  modality: Modality;
  image_urls: string; // Comma-separated image URLs
  created_at: string;
  created_by: string;
  updated_at: string;
  updated_by: string;
}

export interface CreateGeneratorRequest {
  name: string;
  prompt: string;
  provider: string;
  model: string;
  modality: Modality;
  image_urls?: string; // Comma-separated image URLs
}

export interface UpdateGeneratorRequest {
  name: string;
  prompt: string;
  provider: string;
  model: string;
  modality: Modality;
  image_urls?: string; // Comma-separated image URLs
}

export interface GenHistory {
  id: string;
  workspace_id: string;
  generator_id: string;
  request_prompt: string;
  request_model: string;
  request_modality: string;
  request_image_urls: string; // Comma-separated image URLs
  response_content: string;
  response_error: string;
  created_at: string;
  created_by: string;
}

export interface GenerateFromGeneratorRequest {
  generator_id: string;
  prompt: string;
  image_urls?: string[];
}

export interface GenerateFromGeneratorResponse {
  history_id: string;
  content: string;
  error: string;
}

export interface GenModel {
  id: string;
  name: string;
  provider: string;
  provider_display_name: string;
  modality: Modality;
}
